ycmd/completers/go/gocode_completer.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+#
+# Copyright (C) 2015 Google Inc.
+#
+# This file is part of YouCompleteMe.
+#
+# YouCompleteMe is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# YouCompleteMe is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with YouCompleteMe.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+import json
+import subprocess
+
+from ycmd import responses
+from ycmd import utils
+from ycmd.completers.completer import Completer
+
+GO_FILETYPES = set( [ 'go' ] )
+COMPLETION_ERROR_MESSAGE = "gocode call failed"
+PARSE_ERROR_MESSAGE = "gocode result parsing failed"
+NO_COMPLETIONS_MESSAGE = "gocode failed returned no completions"
```

Comment:
Also, please use full sentences for error messages. Capital first letter, punctuation etc.